src/main/kotlin/com/exactpro/th2/common/ConfigurationProviderFactory.kt

```diff
@@ -0,0 +1,61 @@
+/*
+ * Copyright 2020-2022 Exactpro (Exactpro Systems Limited)
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.exactpro.th2.common
+
+import org.apache.commons.cli.CommandLine
+import org.apache.commons.cli.Options
+
+/**
+ * The factory creates an instance of the [ConfigurationProvider] with the specified parameters.
+ *
+ * Often the command line is used to specify these options,
+ * so the factory has methods for parsing the command line
+ */
+interface ConfigurationProviderFactory {
+
+    /**
+     * Class that the factory is responsible for instantiating
+     */
+    val type: Class<out ConfigurationProvider>
+
+    /**
+     * Method for instantiating a [ConfigurationProvider] using parameters.
+     * Each factory implementation can have its own arguments
+     *
+     * To get an array of arguments, use method [parseCommandLine]
+     */
+    fun initProvider(args: Array<String>): ConfigurationProvider
+
+    /**
+     * To parse command line arguments using the apache.commons.cli library,
+     * you must add possible arguments to the [Options] object.
+     * Otherwise, the parser will encounter unknown arguments and throw an exception.
+     *
+     * Since each factory has its own arguments,
+     * this is method for adding arguments to [Options].
+     */
+    fun addOwnOptionsToCmd(options: Options)
+
+    /**
+     * This method allows you to parse the command line object
+     * to get the right parameters in the right order.
+     *
+     * Make sure you called [addOwnOptionsToCmd] before parsing [CommandLine] object.
+     * An exception will be thrown otherwise.
+     */
+    fun parseCommandLine(cmd: CommandLine): Array<String>
```

Comment:
Why do we need those methods?

Reply:
Providers are configured from the command line, and since each provider implementation has its own parameters, factories must be able to parse them.
Configuration example:
```
    val commonFactory = CommonFactory.createFromArguments(
          "--configurationProviderClass",
          "com.exactpro.th2.common.module.provider.FileConfigurationProvider",
          "--file-provider-path",
          "/home/fiodar.rekish/IdeaProjects/th2-common-j/src/test/resources/test_json_configurations",
          "-file-provider-extension",
          "json"
     )
```
These two methods are related to apache cli implementation. 
_addOwnOptionsToCmd_ one will add options to the parser so that it knows what parameters can be and does not fall with the "unknown parameters found" exception. 
_parseCommandLine_ gets these parameters